File: NegocioCSharp/UsuarioConectado.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace LiquidacionSueldos
{
    namespace Negocio
    {
        public partial class UsuarioConectado
        {
            private Datos.Gestor ocdGestor = new Datos.Gestor();
            private DataTable Fila;

            private DataTable Tabla = new DataTable();
            private int _ucoId;
            public int ucoId
            {
                get { return _ucoId; }
                set { _ucoId = value; }
            }

            private System.DateTime _ucoFechaHoraUltimaConexion;
            public System.DateTime ucoFechaHoraUltimaConexion
            {
                get { return _ucoFechaHoraUltimaConexion; }
                set { _ucoFechaHoraUltimaConexion = value; }
            }

            private string _ucoGuid;
            public string ucoGuid
            {
                get { return _ucoGuid; }
                set { _ucoGuid = value; }
            }

            private string _ucoIpPublica;
            public string ucoIpPublica
            {
                get { return _ucoIpPublica; }
                set { _ucoIpPublica = value; }
            }

            private bool _ucoDesconectar;
            public bool ucoDesconectar
            {
                get { return _ucoDesconectar; }
                set { _ucoDesconectar = value; }
            }

            private bool _ucoActivo;
            public bool ucoActivo
            {
                get { return _ucoActivo; }
                set { _ucoActivo = value; }
            }

            private System.DateTime _ucoFechaHoraCreacion;
            public System.DateTime ucoFechaHoraCreacion
            {
                get { return _ucoFechaHoraCreacion; }
                set { _ucoFechaHoraCreacion = value; }
            }

            private System.DateTime _ucoFechaHoraUltimaModificacion;
            public System.DateTime ucoFechaHoraUltimaModificacion
            {
                get { return _ucoFechaHoraUltimaModificacion; }
                set { _ucoFechaHoraUltimaModificacion = value; }
            }

            private int _usuId;
            public int usuId
            {
                get { return _usuId; }
                set { _usuId = value; }
            }

            private int _usuIdCreacion;
            public int usuIdCreacion
            {
                get { return _usuIdCreacion; }
                set { _usuIdCreacion = value; }
            }

            private int _usuIdUltimaModificacion;
            public int usuIdUltimaModificacion
            {
                get { return _usuIdUltimaModificacion; }
                set { _usuIdUltimaModificacion = value; }
            }

            public UsuarioConectado()
            {
                try
                {
                    this.ucoId = 0;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public UsuarioConectado(int ucoId)
            {
                try
                {
                    Fila = new DataTable();
                    Fila = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerUno]", new object[,] { {
                        "@ucoId",
                        ucoId
                    } });

                    if (Fila.Rows.Count > 0)
                    {
                        if (Fila.Rows[0]["ucoId"].ToString().Trim().Length > 0)
                        {
                            this.ucoId = Convert.ToInt32(Fila.Rows[0]["ucoId"]);
                        }
                        else
                        {
                            this.ucoId = 0;
                        }

                        if (Fila.Rows[0]["ucoFechaHoraUltimaConexion"].ToString().Trim().Length > 0)
                        {
                            this.ucoFechaHoraUltimaConexion = Convert.ToDateTime(Fila.Rows[0]["ucoFechaHoraUltimaConexion"]);
                        }
                        else
                        {
                            this.ucoFechaHoraUltimaConexion = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["ucoGuid"].ToString().Trim().Length > 0)
                        {
                            this.ucoGuid = Convert.ToString(Fila.Rows[0]["ucoGuid"]);
                        }
                        else
                        {
                            this.ucoGuid = "";
                        }

                        if (Fila.Rows[0]["ucoIpPublica"].ToString().Trim().Length > 0)
                        {
                            this.ucoIpPublica = Convert.ToString(Fila.Rows[0]["ucoIpPublica"]);
                        }
                        else
                        {
                            this.ucoIpPublica = "";
                        }

                        if (Fila.Rows[0]["ucoFechaHoraCreacion"].ToString().Trim().Length > 0)
                        {
                            this.ucoFechaHoraCreacion = Convert.ToDateTime(Fila.Rows[0]["ucoFechaHoraCreacion"]);
                        }
                        else
                        {
                            this.ucoFechaHoraCreacion = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["ucoFechaHoraUltimaModificacion"].ToString().Trim().Length > 0)
                        {
                            this.ucoFechaHoraUltimaModificacion = Convert.ToDateTime(Fila.Rows[0]["ucoFechaHoraUltimaModificacion"]);
                        }
                        else
                        {
                            this.ucoFechaHoraUltimaModificacion = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["ucoDesconectar"].ToString().Trim().Length > 0)
                        {
                            this.ucoDesconectar = (Convert.ToInt32(Fila.Rows[0]["ucoDesconectar"]) == 1 ? true : false);
                        }
                        else
                        {
                            this.ucoDesconectar = false;
                        }

                        if (Fila.Rows[0]["ucoActivo"].ToString().Trim().Length > 0)
                        {
                            this.ucoActivo = (Convert.ToInt32(Fila.Rows[0]["ucoActivo"]) == 1 ? true : false);
                        }
                        else
                        {
                            this.ucoActivo = false;
                        }

                        if (Fila.Rows[0]["usuId"].ToString().Trim().Length > 0)
                        {
                            this.usuId = Convert.ToInt32(Fila.Rows[0]["usuId"]);
                        }
                        else
                        {
                            this.usuId = 0;
                        }

                        if (Fila.Rows[0]["usuIdCreacion"].ToString().Trim().Length > 0)
                        {
                            this.usuIdCreacion = Convert.ToInt32(Fila.Rows[0]["usuIdCreacion"]);
                        }
                        else
                        {
                            this.usuIdCreacion = 0;
                        }

                        if (Fila.Rows[0]["usuIdUltimaModificacion"].ToString().Trim().Length > 0)
                        {
                            this.usuIdUltimaModificacion = Convert.ToInt32(Fila.Rows[0]["usuIdUltimaModificacion"]);
                        }
                        else
                        {
                            this.usuIdUltimaModificacion = 0;
                        }

                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public UsuarioConectado(int ucoId, System.DateTime ucoFechaHoraUltimaConexion, string ucoGuid, string ucoIpPublica, bool ucoDesconectar, bool ucoActivo, System.DateTime ucoFechaHoraCreacion, System.DateTime ucoFechaHoraUltimaModificacion, int IusuId, int IusuIdCreacion,
            int IusuIdUltimaModificacion)
            {
                try
                {
                    this.ucoId = ucoId;
                    this.ucoFechaHoraUltimaConexion = ucoFechaHoraUltimaConexion;
                    this.ucoGuid = ucoGuid;
                    this.ucoIpPublica = ucoIpPublica;
                    this.ucoDesconectar = ucoDesconectar;
                    this.ucoActivo = ucoActivo;
                    this.ucoFechaHoraCreacion = ucoFechaHoraCreacion;
                    this.ucoFechaHoraUltimaModificacion = ucoFechaHoraUltimaModificacion;
                    this.usuId = usuId;
                    this.usuIdCreacion = usuIdCreacion;
                    this.usuIdUltimaModificacion = usuIdUltimaModificacion;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }


            public DataTable ObtenerConectar(int usuId, string ucoGuid, string ucoIpPublica)
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerConectar]", new object[,] {
                        {
                            "@usuId",
                            usuId
                        },
                        {
                            "@ucoGuid",
                            ucoGuid
                        },
                        {
                            "@ucoIpPublica",
                            ucoIpPublica
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerTodo()
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerTodo]", new object[,] { });
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerTodoCustom()
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerTodoCustom]", new object[,]{});
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerUno(int ucoId)
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerUno]", new object[,] { {
                        "@ucoId",
                        ucoId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public void Actualizar(int ucoId, int usuId, System.DateTime ucoFechaHoraUltimaConexion, string ucoGuid, string ucoIpPublica, bool ucoDesconectar, bool ucoActivo, int usuIdCreacion, int usuIdUltimaModificacion, System.DateTime ucoFechaHoraCreacion,
            System.DateTime ucoFechaHoraUltimaModificacion)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[UsuarioConectado.Actualizar]", new object[,] {
                        {
                            "@ucoId",
                            ucoId
                        },
                        {
                            "@usuId",
                            usuId
                        },
                        {
                            "@ucoFechaHoraUltimaConexion",
                            ucoFechaHoraUltimaConexion
                        },
                        {
                            "@ucoGuid",
                            ucoGuid
                        },
                        {
                            "@ucoIpPublica",
                            ucoIpPublica
                        },
                        {
                            "@ucoDesconectar",
                            ucoDesconectar
                        },
                        {
                            "@ucoActivo",
                            ucoActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@ucoFechaHoraCreacion",
                            ucoFechaHoraCreacion
                        },
                        {
                            "@ucoFechaHoraUltimaModificacion",
                            ucoFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void DesconectarOtros(int usuId, string ucoGuid, string ucoIpPublica)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[UsuarioConectado.DesconectarOtros]", new object[,] {
                        {
                            "@usuId",
                            usuId
                        },
                        {
                            "@ucoGuid",
                            ucoGuid
                        },
                        {
                            "@ucoIpPublica",
                            ucoIpPublica
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Eliminar(int ucoId)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[UsuarioConectado.Eliminar]", new object[,] { {
                        "@ucoId",
                        ucoId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Insertar(int usuId, System.DateTime ucoFechaHoraUltimaConexion, string ucoGuid, string ucoIpPublica, bool ucoDesconectar, bool ucoActivo, int usuIdCreacion, int usuIdUltimaModificacion, System.DateTime ucoFechaHoraCreacion, System.DateTime ucoFechaHoraUltimaModificacion)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[UsuarioConectado.Insertar]", new object[,] {
                        {
                            "@usuId",
                            usuId
                        },
                        {
                            "@ucoFechaHoraUltimaConexion",
                            ucoFechaHoraUltimaConexion
                        },
                        {
                            "@ucoGuid",
                            ucoGuid
                        },
                        {
                            "@ucoIpPublica",
                            ucoIpPublica
                        },
                        {
                            "@ucoDesconectar",
                            ucoDesconectar
                        },
                        {
                            "@ucoActivo",
                            ucoActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@ucoFechaHoraCreacion",
                            ucoFechaHoraCreacion
                        },
                        {
                            "@ucoFechaHoraUltimaModificacion",
                            ucoFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Actualizar()
            {
                try
                {
                    if (this.ucoId != 0)
                    {
                        ocdGestor.EjecutarNonQuery("[UsuarioConectado.Actualizar]", new object[,] {
                            {
                                "@ucoId",
                                ucoId
                            },
                            {
                                "@usuId",
                                usuId
                            },
                            {
                                "@ucoFechaHoraUltimaConexion",
                                ucoFechaHoraUltimaConexion
                            },
                            {
                                "@ucoGuid",
                                ucoGuid
                            },
                            {
                                "@ucoIpPublica",
                                ucoIpPublica
                            },
                            {
                                "@ucoDesconectar",
                                ucoDesconectar
                            },
                            {
                                "@ucoActivo",
                                ucoActivo
                            },
                            {
                                "@usuIdCreacion",
                                usuIdCreacion
                            },
                            {
                                "@usuIdUltimaModificacion",
                                usuIdUltimaModificacion
                            },
                            {
                                "@ucoFechaHoraCreacion",
                                ucoFechaHoraCreacion
                            },
                            {
                                "@ucoFechaHoraUltimaModificacion",
                                ucoFechaHoraUltimaModificacion
                            }
                        });
                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Eliminar()
            {
                try
                {
                    if (this.ucoId != 0)
                    {
                        ocdGestor.EjecutarNonQuery("[UsuarioConectado.Eliminar]", new object[,] { {
                            "@ucoId",
                            ucoId
                        } });
                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public int Insertar()
            {
                int IdMax = 0;
                try
                {
                    IdMax = ocdGestor.EjecutarNonQueryRetornaId("[UsuarioConectado.Insertar]", new object[,] {
                        {
                            "@usuId",
                            usuId
                        },
                        {
                            "@ucoFechaHoraUltimaConexion",
                            ucoFechaHoraUltimaConexion
                        },
                        {
                            "@ucoGuid",
                            ucoGuid
                        },
                        {
                            "@ucoIpPublica",
                            ucoIpPublica
                        },
                        {
                            "@ucoDesconectar",
                            ucoDesconectar
                        },
                        {
                            "@ucoActivo",
                            ucoActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@ucoFechaHoraCreacion",
                            ucoFechaHoraCreacion
                        },
                        {
                            "@ucoFechaHoraUltimaModificacion",
                            ucoFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                return IdMax;
            }

        }
    }
}

File: Negocio/UsuarioConectado.vb
Namespace LiquidacionSueldos
    Namespace Negocio
        Partial Public Class UsuarioConectado
        
			Private ocdGestor As New LiquidacionSueldos.Datos.Gestor
            Private Fila As DataTable
            Private Tabla As New DataTable

			Private _ucoId As Integer
            Public Property ucoId() As Integer
                Get
					Return _ucoId
                End Get
                Set(ByVal value As Integer)
                    _ucoId = value
                End Set
            End Property

			Private _ucoFechaHoraUltimaConexion As Date
            Public Property ucoFechaHoraUltimaConexion() As Date
                Get
					Return _ucoFechaHoraUltimaConexion
                End Get
                Set(ByVal value As Date)
                    _ucoFechaHoraUltimaConexion = value
                End Set
            End Property

			Private _ucoGuid As String
            Public Property ucoGuid() As String
                Get
					Return _ucoGuid
                End Get
                Set(ByVal value As String)
                    _ucoGuid = value
                End Set
            End Property

			Private _ucoIpPublica As String
            Public Property ucoIpPublica() As String
                Get
					Return _ucoIpPublica
                End Get
                Set(ByVal value As String)
                    _ucoIpPublica = value
                End Set
            End Property

			Private _ucoDesconectar As Boolean
            Public Property ucoDesconectar() As Boolean
                Get
					Return _ucoDesconectar
                End Get
                Set(ByVal value As Boolean)
                    _ucoDesconectar = value
                End Set
            End Property

			Private _ucoActivo As Boolean
            Public Property ucoActivo() As Boolean
                Get
					Return _ucoActivo
                End Get
                Set(ByVal value As Boolean)
                    _ucoActivo = value
                End Set
            End Property

			Private _ucoFechaHoraCreacion As Date
            Public Property ucoFechaHoraCreacion() As Date
                Get
					Return _ucoFechaHoraCreacion
                End Get
                Set(ByVal value As Date)
                    _ucoFechaHoraCreacion = value
                End Set
            End Property

			Private _ucoFechaHoraUltimaModificacion As Date
            Public Property ucoFechaHoraUltimaModificacion() As Date
                Get
					Return _ucoFechaHoraUltimaModificacion
                End Get
                Set(ByVal value As Date)
                    _ucoFechaHoraUltimaModificacion = value
                End Set
            End Property

			Private _usuId As Integer
            Public Property usuId() As Integer
                Get
					Return _usuId
                End Get
                Set(ByVal value As Integer)
                    _usuId = value
                End Set
            End Property

			Private _usuIdCreacion As Integer
            Public Property usuIdCreacion() As Integer
                Get
					Return _usuIdCreacion
                End Get
                Set(ByVal value As Integer)
                    _usuIdCreacion = value
                End Set
            End Property

			Private _usuIdUltimaModificacion As Integer
            Public Property usuIdUltimaModificacion() As Integer
                Get
					Return _usuIdUltimaModificacion
                End Get
                Set(ByVal value As Integer)
                    _usuIdUltimaModificacion = value
                End Set
            End Property

			Public Sub New()
                Try                
                    Me.ucoId = 0
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub            

			Public Sub New(ByVal ucoId As Integer)
				Try
                    Fila = New DataTable
                    Fila = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerUno]", New Object(,) {{"@ucoId", ucoId}})

					If Fila.Rows.Count > 0 Then
						If Fila(0)("ucoId").ToString().Trim().Length > 0 Then
                            Me.ucoId = Convert.ToInt32(Fila(0)("ucoId"))
                        Else
                            Me.ucoId = 0
                        End If
                        
						If Fila(0)("ucoFechaHoraUltimaConexion").ToString().Trim().Length > 0 Then
                            Me.ucoFechaHoraUltimaConexion = Convert.ToDateTime(Fila(0)("ucoFechaHoraUltimaConexion"))
                        Else
                            Me.ucoFechaHoraUltimaConexion = Date.Now
                        End If
                        
						If Fila(0)("ucoGuid").ToString().Trim().Length > 0 Then
                            Me.ucoGuid = Convert.ToString(Fila(0)("ucoGuid"))
                        Else
                            Me.ucoGuid = ""
                        End If
                        
						If Fila(0)("ucoIpPublica").ToString().Trim().Length > 0 Then
                            Me.ucoIpPublica = Convert.ToString(Fila(0)("ucoIpPublica"))
                        Else
                            Me.ucoIpPublica = ""
                        End If
                        
						If Fila(0)("ucoFechaHoraCreacion").ToString().Trim().Length > 0 Then
                            Me.ucoFechaHoraCreacion = Convert.ToDateTime(Fila(0)("ucoFechaHoraCreacion"))
                        Else
                            Me.ucoFechaHoraCreacion = Date.Now
                        End If
                        
						If Fila(0)("ucoFechaHoraUltimaModificacion").ToString().Trim().Length > 0 Then
                            Me.ucoFechaHoraUltimaModificacion = Convert.ToDateTime(Fila(0)("ucoFechaHoraUltimaModificacion"))
                        Else
                            Me.ucoFechaHoraUltimaModificacion = Date.Now
                        End If
                        
						If Fila(0)("ucoDesconectar").ToString().Trim().Length > 0 Then
                            Me.ucoDesconectar = IIf(Convert.ToInt32(Fila(0)("ucoDesconectar")) = 1, True, False)
                        Else
							Me.ucoDesconectar = False
                        End If
                        
						If Fila(0)("ucoActivo").ToString().Trim().Length > 0 Then
                            Me.ucoActivo = IIf(Convert.ToInt32(Fila(0)("ucoActivo")) = 1, True, False)
                        Else
							Me.ucoActivo = False
                        End If
                        
						If Fila(0)("usuId").ToString().Trim().Length > 0 Then
                            Me.usuId = Convert.ToInt32(Fila(0)("usuId"))
                        Else
							Me.usuId = 0
                        End If
                        
						If Fila(0)("usuIdCreacion").ToString().Trim().Length > 0 Then
                            Me.usuIdCreacion = Convert.ToInt32(Fila(0)("usuIdCreacion"))
                        Else
							Me.usuIdCreacion = 0
                        End If
                        
						If Fila(0)("usuIdUltimaModificacion").ToString().Trim().Length > 0 Then
                            Me.usuIdUltimaModificacion = Convert.ToInt32(Fila(0)("usuIdUltimaModificacion"))
                        Else
							Me.usuIdUltimaModificacion = 0
                        End If
                        
					End If
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Sub New(ByVal ucoId As Integer, ByVal ucoFechaHoraUltimaConexion As Date, ByVal ucoGuid As String, ByVal ucoIpPublica As String, ByVal ucoDesconectar As Boolean, ByVal ucoActivo As Boolean, ByVal ucoFechaHoraCreacion As Date, ByVal ucoFechaHoraUltimaModificacion As Date, ByVal IusuId As Integer, ByVal IusuIdCreacion As Integer, ByVal IusuIdUltimaModificacion As Integer)
				Try
					me.ucoId = ucoId
					me.ucoFechaHoraUltimaConexion = ucoFechaHoraUltimaConexion
					me.ucoGuid = ucoGuid
					me.ucoIpPublica = ucoIpPublica
					me.ucoDesconectar = ucoDesconectar
					me.ucoActivo = ucoActivo
					me.ucoFechaHoraCreacion = ucoFechaHoraCreacion
					me.ucoFechaHoraUltimaModificacion = ucoFechaHoraUltimaModificacion
					me.usuId = usuId
					me.usuIdCreacion = usuIdCreacion
					me.usuIdUltimaModificacion = usuIdUltimaModificacion
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			
			Public Function ObtenerConectar(ByVal usuId as Integer, ByVal ucoGuid as String, ByVal ucoIpPublica as String) As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerConectar]", New Object(,) {{"@usuId", usuId}, {"@ucoGuid", ucoGuid}, {"@ucoIpPublica", ucoIpPublica}})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerTodo() As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerTodo]", New Object(,) {})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerTodoCustom() As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerTodoCustom]", New Object(,) {})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerUno(ByVal ucoId as Integer) As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[UsuarioConectado.ObtenerUno]", New Object(,) {{"@ucoId", ucoId}})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Sub Actualizar(ByVal ucoId as Integer, ByVal usuId as Integer, ByVal ucoFechaHoraUltimaConexion as Date, ByVal ucoGuid as String, ByVal ucoIpPublica as String, ByVal ucoDesconectar as Boolean, ByVal ucoActivo as Boolean, ByVal usuIdCreacion as Integer, ByVal usuIdUltimaModificacion as Integer, ByVal ucoFechaHoraCreacion as Date, ByVal ucoFechaHoraUltimaModificacion as Date)
				Try
                    ocdGestor.EjecutarNonQuery("[UsuarioConectado.Actualizar]", New Object(,) {{"@ucoId", ucoId}, {"@usuId", usuId}, {"@ucoFechaHoraUltimaConexion", ucoFechaHoraUltimaConexion}, {"@ucoGuid", ucoGuid}, {"@ucoIpPublica", ucoIpPublica}, {"@ucoDesconectar", ucoDesconectar}, {"@ucoActivo", ucoActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@ucoFechaHoraCreacion", ucoFechaHoraCreacion}, {"@ucoFechaHoraUltimaModificacion", ucoFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub DesconectarOtros(ByVal usuId as Integer, ByVal ucoGuid as String, ByVal ucoIpPublica as String)
				Try
                    ocdGestor.EjecutarNonQuery("[UsuarioConectado.DesconectarOtros]", New Object(,) {{"@usuId", usuId}, {"@ucoGuid", ucoGuid}, {"@ucoIpPublica", ucoIpPublica}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Eliminar(ByVal ucoId as Integer)
				Try
                    ocdGestor.EjecutarNonQuery("[UsuarioConectado.Eliminar]", New Object(,) {{"@ucoId", ucoId}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Insertar(ByVal usuId as Integer, ByVal ucoFechaHoraUltimaConexion as Date, ByVal ucoGuid as String, ByVal ucoIpPublica as String, ByVal ucoDesconectar as Boolean, ByVal ucoActivo as Boolean, ByVal usuIdCreacion as Integer, ByVal usuIdUltimaModificacion as Integer, ByVal ucoFechaHoraCreacion as Date, ByVal ucoFechaHoraUltimaModificacion as Date)
				Try
                    ocdGestor.EjecutarNonQuery("[UsuarioConectado.Insertar]", New Object(,) {{"@usuId", usuId}, {"@ucoFechaHoraUltimaConexion", ucoFechaHoraUltimaConexion}, {"@ucoGuid", ucoGuid}, {"@ucoIpPublica", ucoIpPublica}, {"@ucoDesconectar", ucoDesconectar}, {"@ucoActivo", ucoActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@ucoFechaHoraCreacion", ucoFechaHoraCreacion}, {"@ucoFechaHoraUltimaModificacion", ucoFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Actualizar()
                Try
                    If Me.ucoId <> 0 Then
                        ocdGestor.EjecutarNonQuery("[UsuarioConectado.Actualizar]", New Object(,) {{"@ucoId", ucoId}, {"@usuId", usuId}, {"@ucoFechaHoraUltimaConexion", ucoFechaHoraUltimaConexion}, {"@ucoGuid", ucoGuid}, {"@ucoIpPublica", ucoIpPublica}, {"@ucoDesconectar", ucoDesconectar}, {"@ucoActivo", ucoActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@ucoFechaHoraCreacion", ucoFechaHoraCreacion}, {"@ucoFechaHoraUltimaModificacion", ucoFechaHoraUltimaModificacion}})
                    End If
                Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Sub Eliminar()
                Try
                    If Me.ucoId <> 0 Then
                        ocdGestor.EjecutarNonQuery("[UsuarioConectado.Eliminar]", New Object(,) {{"@ucoId", ucoId}})
                    End If
                Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Function Insertar() as Integer
                Dim IdMax As Integer
                Try
                    IdMax = ocdGestor.EjecutarNonQueryRetornaId("[UsuarioConectado.Insertar]", New Object(,) {{"@usuId", usuId}, {"@ucoFechaHoraUltimaConexion", ucoFechaHoraUltimaConexion}, {"@ucoGuid", ucoGuid}, {"@ucoIpPublica", ucoIpPublica}, {"@ucoDesconectar", ucoDesconectar}, {"@ucoActivo", ucoActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@ucoFechaHoraCreacion", ucoFechaHoraCreacion}, {"@ucoFechaHoraUltimaModificacion", ucoFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
                Return IdMax
            End Function

		End Class
    End Namespace
End Namespace